EMS/client-helper-bundle/src/Security/CoreApi/CoreApiAuthenticator.php

```diff
@@ -0,0 +1,92 @@
+<?php
+
+declare(strict_types=1);
+
+namespace EMS\ClientHelperBundle\Security\CoreApi;
+
+use EMS\ClientHelperBundle\Security\CoreApi\User\CoreApiUserProvider;
+use EMS\ClientHelperBundle\Security\Login\LoginCredentials;
+use EMS\ClientHelperBundle\Security\Login\LoginForm;
+use EMS\CommonBundle\Contracts\CoreApi\Exception\NotAuthenticatedExceptionInterface;
+use Psr\Log\LoggerInterface;
+use Symfony\Component\Form\FormFactory;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\Security\Core\Authentication\Token\TokenInterface;
+use Symfony\Component\Security\Core\Exception\AuthenticationException;
+use Symfony\Component\Security\Core\Security;
+use Symfony\Component\Security\Http\Authenticator\AbstractAuthenticator;
+use Symfony\Component\Security\Http\Authenticator\Passport\Badge\CsrfTokenBadge;
+use Symfony\Component\Security\Http\Authenticator\Passport\Badge\UserBadge;
+use Symfony\Component\Security\Http\Authenticator\Passport\Passport;
+use Symfony\Component\Security\Http\Authenticator\Passport\SelfValidatingPassport;
+use Symfony\Component\Security\Http\HttpUtils;
+use Symfony\Component\Security\Http\Util\TargetPathTrait;
+
+class CoreApiAuthenticator extends AbstractAuthenticator
+{
+    use TargetPathTrait;
+
+    public const CSRF_ID = 'login';
+
+    public function __construct(
+        private readonly HttpUtils $httpUtils,
+        private readonly CoreApiFactory $coreApiFactory,
+        private readonly CoreApiUserProvider $coreApiUserProvider,
+        private readonly FormFactory $formFactory,
+        private readonly LoggerInterface $logger,
+        private readonly string $routeLogin,
+    ) {
+    }
+
+    public function supports(Request $request): ?bool
+    {
+        return $request->isMethod(Request::METHOD_POST) && $request->get('_route') === $this->routeLogin;
+    }
+
+    public function authenticate(Request $request): Passport
+    {
+        $csrfToken = $request->request->get('token');
```

Comment:
Does the csrf needs validations?

Reply:
It will be validated even twice

-> by the form with  $form->handleRequest($request);
-> with the new CsrfTokenBadge(self::CSRF_ID, $csrfToken) on line 75